cmd/watcher/main.go

```diff
@@ -105,3 +122,26 @@ func connectToAPIServer(ctx context.Context, apiAddr string, authMode string) (*
 	defer cancel()
 	return grpc.DialContext(ctx, apiAddr, opts...)
 }
+
+func loadCerts() (*x509.CertPool, error) {
+	// Setup TLS certs to the server.
+	f, err := os.Open("/etc/tls/tls.crt")
+	if err != nil {
+		log.Println("no local cluster cert found, defaulting to system pool...")
+		return x509.SystemCertPool()
+	}
+	defer f.Close()
```

Comment:
nit: shouldn't this defer be before the err != nil line so that if we hit the err block, we still call f.Close before returning?

Reply:
Doesn't make a difference. If err is returned then [f will be nil](https://go.googlesource.com/go/+/go1.15.7/src/os/file.go#329), so the [close won't do anything and return an error](https://go.googlesource.com/go/+/go1.15.7/src/os/file_posix.go#21).